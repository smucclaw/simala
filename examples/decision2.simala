let
  
  get_opinion =

    fun (b, inv) =>
      let opaque `Seed / IT / Pre-revenue condition` =
            b.stage     == 'seed
         && b.sector    == 'information_technology
         && b.stage_com == 'pre_revenue
      in
      let opaque `Series A / IT / Pre-profit condition` =
            b.stage     == 'series_a
         && b.sector    == 'information_technology
         && b.stage_com == 'pre_profit
      in
      let opaque `has_ESG condition` =
            inv.has_ESG
      in
      let opaque `overall choice` =
           `Seed / IT / Pre-revenue condition`
        || `Series A / IT / Pre-profit condition`
        || `has_ESG condition`
      in
      let opaque result =
        if `overall choice` then 'interesting else 'reject
      in
        result

in let

  business =
    { stage     = 'seed,
      sector    = 'information_technology,
      stage_com = 'pre_profit
    }

in let

  investor =
    { has_ESG = true }

in

  get_opinion(business, investor)
