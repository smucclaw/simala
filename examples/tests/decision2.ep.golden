get_opinion =

  fun (b, inv) =>
    let opaque `Seed / IT / Pre-revenue condition` =
          b.stage     == 'seed
       && b.sector    == 'information_technology
       && b.stage_com == 'pre_revenue
    in
    let opaque `Series A / IT / Pre-profit condition` =
          b.stage     == 'series_a
       && b.sector    == 'information_technology
       && b.stage_com == 'pre_profit
    in
    let opaque `has_ESG condition` =
          inv.has_ESG
    in
    let opaque `overall choice` =
         `Seed / IT / Pre-revenue condition`
      || `Series A / IT / Pre-profit condition`
      || `has_ESG condition`
    in
    let opaque result =
      if `overall choice` then 'interesting else 'reject
    in
      result ;

business =
  { stage     = 'seed,
    sector    = 'information_technology,
    stage_com = 'pre_profit
  } ;

investor =
  { has_ESG = true } ;

#eval get_opinion(business, investor)
